Pinta.Effects/Effects/JuliaFractalEffect.cs

```diff
@@ -42,76 +42,97 @@ public override void LaunchConfiguration ()
 	private static double Julia (double x, double y, double r, double i)
 	{
 		double c = 0;
-
-		while (c < 256 && x * x + y * y < 10000) {
+		while (c < 256 && ((x * x) + (y * y) < 10000)) {
 			double t = x;
-			x = x * x - y * y + r;
-			y = 2 * t * y + i;
+			x = (x * x) - (y * y) + r;
+			y = (2 * t * y) + i;
 			++c;
 		}
-
-		c -= 2 - 2 * log2_10000 / Math.Log (x * x + y * y);
-
-		return c;
+		return c - (2 - 2 * log2_10000 / Math.Log ((x * x) + (y * y)));
 	}
 
 	public override void Render (ImageSurface src, ImageSurface dst, ReadOnlySpan<RectangleI> rois)
 	{
-		const double jr = 0.3125;
-		const double ji = 0.03;
-
-		int w = dst.Width;
-		int h = dst.Height;
-		double invH = 1.0 / h;
-		double invZoom = 1.0 / Data.Zoom;
-		double invQuality = 1.0 / Data.Quality;
-		double aspect = (double) h / (double) w;
-		int count = Data.Quality * Data.Quality + 1;
-		double invCount = 1.0 / (double) count;
-		double angleTheta = (Data.Angle * Math.PI * 2) / 360.0;
-
+		JuliaSettings settings = CreateSettings (dst);
 		Span<ColorBgra> dst_data = dst.GetPixelData ();
-		int dst_width = dst.Width;
-
 		foreach (Core.RectangleI rect in rois) {
 			for (int y = rect.Top; y <= rect.Bottom; y++) {
-				var dst_row = dst_data.Slice (y * dst_width, dst_width);
-
+				var dst_row = dst_data.Slice (y * settings.w, settings.w);
 				for (int x = rect.Left; x <= rect.Right; x++) {
-					int r = 0;
-					int g = 0;
-					int b = 0;
-					int a = 0;
+					LoopContext context = new (y, x);
+					dst_row[x] = GetPixelColor (settings, context);
+				}
+			}
+		}
+	}
+
+	private readonly record struct LoopContext (int y, int x);
```

Comment:
This struct is unnecessary IMO - it's just an x,y coordinate so I think it's reasonable to pass those straight into `GetPixelColor` directly or use a `PointI`

Reply:
Yeah, I can see your point. I removed `LoopContext` from both PRs and I'm now using `PointI` instead